Sources/SourceKitLSP/Workspace.swift

```diff
@@ -255,6 +256,23 @@ package final class Workspace: Sendable {
     await syntacticTestIndex.filesDidChange(events)
     await semanticIndexManager?.filesDidChange(events)
   }
+
+  func documentService(for uri: DocumentURI) -> LanguageService? {
+    return documentService.value[uri]
+  }
+
+  /// Set a language service for a document uri only if it doesn't already exist.
+  /// - Returns: The language service of the given `uri`
+  func documentService(for uri: DocumentURI, newLanguageService: any LanguageService) -> LanguageService {
```

Comment:
At its core, this method is a setter but its current name sounds like a getter. What do you think about?

```suggestion
  /// Set a language service for a document uri and returns if none exists already.
  /// If a language service already exists for this document, eg. because two requests start creating a language service for a document and race, `newLanguageService` is dropped and the existing language service for the document is returned.
  func setDocumentService(for uri: DocumentURI, _ newLanguageService: any LanguageService) -> LanguageService {
```